### ISSUE TITLE
PSR2.Methods.FunctionCallSignature.Indent error with multiline short closure returns

### ISSUE THREAD

**Issue by sspat** [squizlabs/PHP_CodeSniffer#2571]
# Versions
codesniffer@dev-master
php 7.4.0beta1

# Description
When adding a short syntax closure which has a multiline return statement an incorrect indent error is raised, if the return statement begins on the same line as the closure declaration.

```php
<?php

declare(strict_types=1);

$container
    ->add(
        'identifier',
        fn() : SomeClass => new SomeClass(
            [
                'option' => true,
                'option2' => false,
            ]
        ),
        true
    );

```
```
9 | ERROR | [x] Multi-line function call not indented correctly; expected 8 spaces but found 12
   |       |     (PSR2.Methods.FunctionCallSignature.Indent
```

However, there is no error when indenting like this:
```php
<?php

declare(strict_types=1);

$container
    ->add(
        'identifier',
        fn() : SomeClass =>
        new SomeClass(
            [
                'option' => true,
                'option2' => false,
            ]
        ),
        true
    );
```

# Expected behavior
Did a lookup on PSR-2/PSR-12 but found nothing regarding short closure syntax.
In my opinion, the first example looks better, despite the fact, that the second is more consistent with full closure indenting rules:
```php
<?php

declare(strict_types=1);

$container
    ->add(
        'identifier',
        function () : SomeClass {
            return new SomeClass(
                [
                    'option' => true,
                    'option2' => false,
                ]
            );
        },
        true
    );

```
As there is no explicit rule for short closure identation in PSR2, maybe this sniff could make an exception for short closures and ignore them?

**Comment by gsherwood** [comment]
Closing as merged with #2523
